pkg/exchange/max/exchange.go

```diff
@@ -270,18 +270,34 @@ func (e *Exchange) QueryOpenOrders(ctx context.Context, symbol string) (orders [
 	return orders, err
 }
 
-// lastOrderID is not supported on MAX
 func (e *Exchange) QueryClosedOrders(
 	ctx context.Context, symbol string, since, until time.Time, lastOrderID uint64,
 ) ([]types.Order, error) {
-	log.Warn("!!!MAX EXCHANGE API NOTICE!!! the since/until conditions will not be effected on closed orders query, max exchange does not support time-range-based query")
 	if !since.IsZero() || !until.IsZero() {
-		return e.queryClosedOrdersByTime(ctx, symbol, since, until)
+		return e.queryClosedOrdersByTime(ctx, symbol, since, until, maxapi.OrderByAsc, 1000)
 	}
 
 	return e.queryClosedOrdersByLastOrderID(ctx, symbol, lastOrderID)
 }
 
+func (e *Exchange) QueryClosedOrdersDesc(
+	ctx context.Context, symbol string, since, until time.Time, lastOrderID uint64,
+) ([]types.Order, error) {
+	closedOrders, err := e.queryClosedOrdersByTime(ctx, symbol, since, until, maxapi.OrderByDesc, 1000)
+	if lastOrderID == 0 {
+		return closedOrders, err
+	}
+
+	var filterClosedOrders []types.Order
+	for _, closedOrder := range filterClosedOrders {
+		if closedOrder.OrderID > lastOrderID {
```

Comment:
should be compared only when lastOrderID is not zero

Reply:
Yes, check lastOrderID at line 287.